algorithms/search/next_greatest_letter.py

```diff
@@ -0,0 +1,37 @@
+'''
+Given a list of sorted characters letters containing only lowercase letters,
+and given a target letter target, find the smallest element in the list that
+is larger than the given target.
+
+Letters also wrap around. For example, if the target is target = 'z' and
+letters = ['a', 'b'], the answer is 'a'.
+
+Input:
+letters = ["c", "f", "j"]
+target = "a"
+Output: "c"
+
+Input:
+letters = ["c", "f", "j"]
+target = "c"
+Output: "f"
+
+Input:
+letters = ["c", "f", "j"]
+target = "d"
+Output: "f"
+
+Reference: https://leetcode.com/problems/find-smallest-letter-greater-than-target/description/
+'''
+
+import bisect
+
+def next_greatest_letter(letters, target):
+    index = bisect.bisect(letters, target)
+    return letters[index % len(letters)]
+
+def next_greatest_letter_v2(letters, target):
```

Comment:
It will good to mention the time complexities here, as this one is linear.  
And maybe we should add a third version with `log n` with manual binary search (without using `bisect`)??